file_catalog/mongo.py

```diff
@@ -93,134 +112,207 @@ def find_files(self, query={}, keys=None, limit=None, start=0):
-        result = self.client.files.update_one({'uuid': uuid},
-                                              {'$set': metadata})
+    def update_file(self, uuid: str, metadata: Dict[str, Any]) -> None:
+        """Update file."""
+        result = self.client.files.update_one({"uuid": uuid}, {"$set": metadata})
 
         if result.modified_count is None:
-            logger.warn('Cannot determine if document has been modified since `result.modified_count` has the value `None`. `result.matched_count` is %s' % result.matched_count)
+            logger.warning(
+                "Cannot determine if document has been modified since `result.modified_count` has the value `None`. `result.matched_count` is %s",
+                result.matched_count,
+            )
         elif result.modified_count != 1:
-            logger.warn('updated %s files with id %r',
-                        result.modified_count, uuid)
-            raise Exception('did not update')
+            logger.warning("updated %s files with id %r", result.modified_count, uuid)
+            raise Exception("did not update")
 
     @run_on_executor
-    def replace_file(self, metadata):
-        uuid = metadata['uuid']
+    def replace_file(self, metadata: Dict[str, Any]) -> None:
+        """Replace file.
 
-        result = self.client.files.replace_one({'uuid': uuid},
-                                               metadata)
+        Metadata must include 'uuid'.
+        """
+        uuid = metadata["uuid"]
+
+        result = self.client.files.replace_one({"uuid": uuid}, metadata)
 
         if result.modified_count is None:
-            logger.warn('Cannot determine if document has been modified since `result.modified_count` has the value `None`. `result.matched_count` is %s' % result.matched_count)
+            logger.warning(
+                "Cannot determine if document has been modified since `result.modified_count` has the value `None`. `result.matched_count` is %s",
+                result.matched_count,
+            )
         elif result.modified_count != 1:
-            logger.warn('updated %s files with id %r',
-                        result.modified_count, uuid)
-            raise Exception('did not update')
+            logger.warning("updated %s files with id %r", result.modified_count, uuid)
+            raise Exception("did not update")
 
     @run_on_executor
-    def delete_file(self, filters):
+    def delete_file(self, filters: Dict[str, Any]) -> None:
+        """Delete file matching filters."""
         result = self.client.files.delete_one(filters)
 
         if result.deleted_count != 1:
-            logger.warn('deleted %d files with filter %r',
-                        result.deleted_count, filter)
-            raise Exception('did not delete')
+            logger.warning(
+                "deleted %d files with filter %r", result.deleted_count, filters
+            )
+            raise Exception("did not delete")
```

Comment:
`"did not delete"` doesn't match so well with the logging statement above it.
Consider building a string and using it in both the logger warning and the exception message?

Reply:
sure thing